Mono.Debugging.Soft/SoftEvaluationContext.cs

```diff
@@ -184,7 +185,11 @@ Value RuntimeInvoke (MethodMirror method, object target, Value[] values, bool en
 				return method.Evaluate (target is TypeMirror ? null : (Value) target, values);
 			} catch (NotSupportedException) {
 				AssertTargetInvokeAllowed ();
-
+				var threadState = Thread.ThreadState;
+				if (threadState == ThreadState.WaitSleepJoin) {
```

Comment:
This should be `if ((threadState & ThreadState.WaitSleepJoin) == ThreadState.WaitSleepJoin) {`